lib/msf/core/post/linux/kernel.rb

```diff
@@ -1,331 +1,338 @@
+          return false if read_file('/proc/sys/kernel/unprivileged_userns_clone').to_s.strip.eql? '0'
+
+          true
+        rescue StandardError
+          raise 'Could not determine userns status'
+        end
+
+        #
+        # Returns true if Address Space Layout Randomization (ASLR) is enabled
+        #
+        # @return [Boolean]
+        #
+        def aslr_enabled?
+          aslr = read_file('/proc/sys/kernel/randomize_va_space').to_s.strip
+          aslr.eql?('1') || aslr.eql?('2')
+        rescue StandardError
+          raise 'Could not determine ASLR status'
+        end
+
+        #
+        # Returns true if Exec-Shield is enabled
+        #
+        # @return [Boolean]
+        #
+        def exec_shield_enabled?
+          exec_shield = read_file('/proc/sys/kernel/exec-shield').to_s.strip
+          exec_shield.eql?('1') || exec_shield.eql?('2')
+        rescue StandardError
+          raise 'Could not determine exec-shield status'
+        end
+
+        #
+        # Returns true if unprivileged bpf is disabled
+        #
+        # @return [Boolean]
+        #
+        def unprivileged_bpf_disabled?
+          unprivileged_bpf_disabled = read_file('/proc/sys/kernel/unprivileged_bpf_disabled').to_s.strip
+          return unprivileged_bpf_disabled == '1' || unprivileged_bpf_disabled == '2'
+        rescue StandardError
+          raise 'Could not determine kernel.unprivileged_bpf_disabled status'
+        end
+
+        #
+        # Returns true if kernel pointer restriction is enabled
+        #
+        # @return [Boolean]
+        #
+        def kptr_restrict?
+          read_file('/proc/sys/kernel/kptr_restrict').to_s.strip.eql? '1'
+        rescue StandardError
+          raise 'Could not determine kernel.kptr_restrict status'
+        end
+
+        #
+        # Returns true if dmesg restriction is enabled
+        #
+        # @return [Boolean]
+        #
+        def dmesg_restrict?
+          read_file('/proc/sys/kernel/dmesg_restrict').to_s.strip.eql? '1'
+        rescue StandardError
+          raise 'Could not determine kernel.dmesg_restrict status'
+        end
+
+        #
+        # Returns mmap minimum address
+        #
+        # @return [Integer]
+        #
+        def mmap_min_addr
+          mmap_min_addr = read_file('/proc/sys/vm/mmap_min_addr').to_s.strip
+          return 0 unless mmap_min_addr =~ /\A\d+\z/
+
+          mmap_min_addr
+        rescue StandardError
+          raise 'Could not determine system mmap_min_addr'
+        end
+
+        #
+        # Returns true if Linux Kernel Runtime Guard (LKRG) kernel module is installed
+        #
+        def lkrg_installed?
+          directory?('/proc/sys/lkrg')
+        rescue StandardError
+          raise 'Could not determine LKRG status'
+        end
+
+        #
+        # Returns true if grsecurity is installed
+        #
+        def grsec_installed?
+          cmd_exec('test -c /dev/grsec && echo true').to_s.strip.include? 'true'
```

Comment:
Would it be worth adding a function like `character_file?` in `file.rb` similar to: https://github.com/rapid7/metasploit-framework/blob/0cf227ff00e655fbe12a100959272b1e7490f362/lib/msf/core/post/file.rb#L186
for handling this scenario?

Reply:
@bcoles wrote that, I simply reverted a bad change to it so it works again

Reply:
@h00die sounds good. It doesn't seem like this method of checking for a character file is used anywhere else in the framework. We can refactor later if need be.